agents-api/agents_api/activities/execute_integration.py

```diff
@@ -54,32 +55,61 @@ async def execute_integration(
     setup = merged_tool_setup.get(tool_name, {}) | (integration.setup or {}) | setup
 
     try:
+        # Handle dummy provider as a special case
         if integration.provider == "dummy":
             return arguments
 
+        # Call the integration service
         integration_service_response = await integrations.run_integration_service(
             provider=integration.provider,
             setup=setup,
             method=integration.method,
             arguments=arguments,
         )
 
+        # Check for error in the response
         if integration_service_response.get("error"):
+            error_message = integration_service_response["error"]
+            integration_str = (
+                f"{integration.provider}.{integration.method}"
+                if integration.method
```

Comment:
what about edge cases, if `integration.method` or `integration.provider` is a empty string ?

Reply:
cant be coz of typespec definitions restrict them to be specific values